### ISSUE TITLE
CRTP should support multiple overloads of a method

### ISSUE THREAD

**Issue by nodakai** [fsharp/fslang-suggestions#1342]
**I propose we** enable compilation of the following CRTP function:

```f#
let inline f(t: ^T when ^T: (member Foo: int -> int) and ^T: (member Foo: float -> float)) =                                                                                                                                     
  printfn "%d" <| (^T: (member Foo: int -> int) (t, 0))                                                                                                                                                                          
  printfn "%f" <| (^T: (member Foo: float -> float) (t, 0.)) 
```

(This could also be adapted to the simplified syntax introduced in F# 7)

**The existing way of approaching this problem in F# is** to handle reference types or immutable structs by passing them as two separate arguments, each with its own CRTP type parameter. However, this approach is not feasible for mutable structs or structs following the builder pattern.

## Pros and Cons 

**The advantages of making this adjustment to F# are** primarily enhancing the language's consistency and expanding its capabilities.

**The disadvantages of making this adjustment to F# are** potentially introducing more complexity into the compiler's implementation.

## Extra information

**Estimated cost (XS, S, M, L, XL, XXL):**  Idk, maybe L?

**Related suggestions:** (put links to related suggestions here)

## Affidavit (please submit!)

Please tick these items by placing a cross in the box:
* [v] This is not a question (e.g. like one you might ask on [StackOverflow](http://stackoverflow.com)) and I have searched StackOverflow for discussions of this issue
* [v] This is a language change and not purely a tooling change (e.g. compiler bug, editor support, warning/error messages, new warning, non-breaking optimisation) belonging to [the compiler and tooling repository](https://github.com/dotnet/fsharp)
* [v] This is not something which has obviously "already been decided" in previous versions of F#. If you're questioning a fundamental design decision that has obviously already been taken (e.g. "Make F# untyped") then please don't submit it
* [v] I have [searched both open and closed suggestions on this site](http://github.com/fsharp/fslang-suggestions/issues) and believe this is not a duplicate

Please tick all that apply:
* [v] This is not a breaking change to the F# language design
* [ ] I or my company would be willing to help implement and/or test this -- I'm willing to test, lack necessary bandwidth and expertise to implement it by myself

## For Readers

If you would like to see this issue implemented, please click the :+1: emoji on this issue. These counts are used to generally order the suggestions by engagement.

**Comment by voronoipotato** [comment]
yeah I think one of the ways people have hacked around this is by adding dummy arguments to the end. It's very stupid, but it does seem to work. 

```fsharp
type ICounts() =
     static member inline Count< ^C when ^C: (member itemCount: int)> o =  (^C: (member itemCount: int) o)
     static member inline Count< ^C, 'Dummy1 when ^C: (member thingCount: int)> o =  (^C: (member thingCount: int) o)
     static member inline Count< ^C, 'Dummy1, 'Dummy2 when ^C: (member otherCount: int)> o =  (^C: (member otherCount: int) o)
     static member inline Count< ^C, 'Dummy1, 'Dummy2, 'Dummy3 when ^C: (member Count: int)> o =  (^C: (member Count: int) o)

let say count = 
    printfn "%d" count
    ()

type c = ICounts

say (c.Count {|itemCount = 1|})
say (c.Count {|thingCount = 2|})
say (c.Count {|otherCount = 3|})
say (c.Count {|Count = 4|})
```
